### PR TITLE
Amend gulp express task to use next available port

### PR DESCRIPTION
**Pull request by myabc**
If a given port is in use, increment.
Also, make Protractor aware of the baseUrl.

_This should fix problems running multiple jobs on internal CI slaves (#2217 didn't accomplish this)._

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-gulp-increment-express-port.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: gulpfile.js [hunk 1]
```diff
@@ -72,10 +72,21 @@ gulp.task('sass', function() {
 });
 
 gulp.task('express', function() {
-  var app = require('./protractor/server');
-  server = app.listen(8080, function() {
-    console.log('Listening at localhost:8080');
-  });
+  var expressApp = require('./protractor/server');
+  var port = process.env.PORT || 8080;
```
Comment (ulferts) [inline gulpfile.js:76]: Why not take a random port than (within an interval. Should lead to fewer clashes)

Comment (myabc) [inline gulpfile.js:76]: @ulferts this server is also useful for front-end development (e.g. `gulp`'s default task). A non-random default makes life easier for developers.

Comment (ulferts) [inline gulpfile.js:76]: OK

**Pull request merged**